server/src/main/java/org/opensearch/index/mapper/BinaryFieldMapper.java

```diff
@@ -240,30 +240,28 @@ protected String contentType() {
      */
     public static class CustomBinaryDocValuesField extends CustomDocValuesField {
 
-        private final ObjectArrayList<byte[]> bytesList;
-
-        private int totalSize = 0;
+        private final ArrayList<byte[]> bytesList;
 
         public CustomBinaryDocValuesField(String name, byte[] bytes) {
             super(name);
-            bytesList = new ObjectArrayList<>();
+            bytesList = new ArrayList<>();
             add(bytes);
         }
 
         public void add(byte[] bytes) {
             bytesList.add(bytes);
-            totalSize += bytes.length;
         }
 
         @Override
         public BytesRef binaryValue() {
             try {
-                CollectionUtils.sortAndDedup(bytesList);
-                int size = bytesList.size();
-                BytesStreamOutput out = new BytesStreamOutput(totalSize + (size + 1) * 5);
-                out.writeVInt(size);  // write total number of values
-                for (int i = 0; i < size; i++) {
-                    final byte[] value = bytesList.get(i);
+                // sort and dedup in place
+                CollectionUtils.sortAndDedup(bytesList, Arrays::compareUnsigned);
+                int size = bytesList.stream().map(b -> b.length).reduce(0, Integer::sum);
```

Comment:
`bytesList.stream().mapToInt(b -> b.length).sum()` is slightly simpler and makes the intent more clear I think